src/tools/illink/test/Mono.Linker.Tests.Cases/DataFlow/ExceptionalDataFlow.cs

```diff
@@ -730,6 +754,32 @@ public static void NestedFinally ()
 			}
 		}
 
+		[ExpectedWarning ("IL2072", nameof (RequireAll1) + "(Type)", nameof (GetWithPublicMethods) + "()")]
+		[ExpectedWarning ("IL2072", nameof (RequireAll2) + "(Type)", nameof (GetWithPublicMethods) + "()")]
+		[ExpectedWarning ("IL2072", nameof (RequireAll3) + "(Type)", nameof (GetWithPublicFields) + "()")]
+		[ExpectedWarning ("IL2072", nameof (RequireAll4) + "(Type)", nameof (GetWithPublicFields) + "()")]
+
+		// Trimmer merges branches going forward.
+		[ExpectedWarning ("IL2072", nameof (RequireAll3) + "(Type)", nameof (GetWithPublicMethods) + "()",
+			ProducedBy = Tool.Trimmer | Tool.NativeAot)]
+		[ExpectedWarning ("IL2072", nameof (RequireAll4) + "(Type)", nameof (GetWithPublicMethods) + "()",
+			ProducedBy = Tool.Trimmer | Tool.NativeAot)]
+		public static void ChangeInFinallyNestedInFinally ()
+		{
+			Type t = GetWithPublicMethods ();
+			try {
+				RequireAll1 (t);
+			} finally {
+				try {
+					RequireAll2 (t);
+				} finally {
+					t = GetWithPublicFields ();
+				}
+				RequireAll3 (t); // fields only
```

Comment:
Just to confirm - the warning was raised when processing the finally exception state, right? The "normal" state path didn't produce the warning because it already propagates it through predecessor's finally blocks.